File: Parser/NurseParser.cs
using System;
using Domain.Models;
using Parser.Parsers;

namespace Parser;

public class NurseParser : IParser
{
	public NurseParser()
	{
	}

    public Schedule ParseFromTxt(string textFile)
    {
        //change this later
        return IdiotParser.Parse();
    }

    public Schedule ParseFromXML(string xml)
    {
        return XMLParser.Parse(xml);
    }

    public Schedule ParseFromJSON(string json)
    {
        return JSONParser.Parse(json);
    }


}

File: Parser/NurseParser.py
from Parser.Parsers.JSONParser import JSONParser

import json


class NurseParser:
    def __init__(self):
        self.jsonParser = JSONParser()

    def parseFromJSON(self, filePath):
        file = open(filePath)
        jsonData = json.load(file)
        parsedData = self.jsonParser.parse(jsonData)
        return parsedData

    def parseScenario(self, scenario, example=False):
        return self.jsonParser.parseNRC(scenario, example)
